src/extensions/extra/selfie-taker/camera.js

```diff
@@ -0,0 +1,201 @@
+/*jslint vars: true, plusplus: true, devel: true, nomen: true,  regexp: true, indent: 4, maxerr: 50 */
+/*global define, brackets, $, window */
+
+define(function (require, exports, module) {
+    "use strict";
+
+    var Dialog = brackets.getModule("widgets/Dialogs");
+    var Buffer = brackets.getModule("filesystem/impls/filer/BracketsFiler").Buffer;
+    var Strings = brackets.getModule("strings");
+    var fs = brackets.getModule("fileSystemImpl");
+    var selfieDialogHTML = require("text!dialog.html");
+    var selfieWidgetHTML = require("text!selfieWidget.html");
+    var _selfieDialog;
+    var video;
+    var photo;
+    var snap;
+    var snapAudio;
+    var canvas;
+    var use;
+    var deferred;
+    var filePath;
+    var streaming = false;
+    var height;
+    var width = 320;
+    var vendorURL = window.URL || window.webkitURL;
+    var navigator = window.navigator;
+    navigator.getMedia =   (navigator.getUserMedia ||
+                            navigator.webkitGetUserMedia ||
+                            navigator.mozGetUserMedia ||
+                            navigator.msGetUserMedia);
+
+
+    /**
+     * Dialog Buttons IDs
+     * @const {string}
+     */
+    var DIALOG_BTN_CANCEL = "cancel";
+    var DIALOG_BTN_OK = "ok";
+
+    /**
+     * Dialog Buttons Class Names
+     * @const {string}
+     */
+    var DIALOG_BTN_CLASS_PRIMARY = "primary";
+    var DIALOG_BTN_CLASS_LEFT = "left";
+
+    // Based on http://stackoverflow.com/questions/21797299/convert-base64-string-to-arraybuffer
+    function base64ToBuffer(base64) {
+        // This code will have to change to use Filer's Buffer api
+        var bin = window.atob(base64);
```

Comment:
Only if this is not valid base64 data. But since we control that (we use a regex before passing it to this function, if anything, it will throw there), there's probably no need to try...catch